tests/diff/document_validation_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// TestDocumentValidation tests validation rules for documents insert and update.
+func TestDocumentValidation(t *testing.T) {
+	t.Parallel()
+	ctx, db := setup(t)
+
+	// initiate a collection with a valid document, so we have something to update
+	collection := db.Collection("document-validation")
+	_, err := collection.InsertOne(ctx, bson.D{
+		{"_id", "valid"},
+		{"v", int32(42)},
+	})
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct {
+		expectedErrMongoDB  *mongo.CommandError
+		expectedErrFerretDB *mongo.CommandError
```

Comment:
Notice how https://github.com/FerretDB/dance/blob/main/tests/diff/names_test.go and other tests don't have different expected errors. That's because of our test configuration at https://github.com/FerretDB/dance/blob/main/tests/diff.yml. See also https://github.com/FerretDB/dance/blob/main/CONTRIBUTING.md#running-tests